### ISSUE TITLE
System Spider Cache expires after 24 hours

### ISSUE THREAD

**Issue by guest1604** [TACC/Lmod#344]
Hello,

I use Lmod 7.7 configured with System Spider Cache, and I don’t understand why the cache expires after 24h.

Below, my configuration :

```
$ module --version
Modules based on Lua: Version 7.7  2017-09-19 11:05 -05:00

    by Robert McLay mclay@tacc.utexas.edu
``` 
``` 
$ cat /home/user/lmodrc.lua
(...)
scDescriptT = {
  {
    ["dir"]       = "/home/user/moduleData/cacheDir",
    ["timestamp"] = "/home/user/moduleData/system.txt",
  },
}
``` 
```  
$ module --config
(...)
Active RC file(s):
------------------
/usr/share/lmod/lmod/init/lmodrc.lua
/home/user/lmodrc.lua

Cache Directory                 Time Stamp File
---------------                 ---------------
/home/user/moduleData/cacheDir  /home/user/moduleData/system.txt
(...)
``` 

Below, some tests :
``` 
$ cd /home/user/moduleData
$ find -type f -ls
23055583  496 -rw-r--r--   1 user user        501501 févr.  9 15:16 ./cacheDir/spiderT.lua
23055579  364 -rw-r--r--   1 user user        365877 févr.  9 15:16 ./cacheDir/spiderT.luac_5.1
32385948    0 -rw-r--r--   1 user user             0 févr.  9 15:15 ./system.txt

$ export LMOD_ANCIENT=1000000000
$ rm -rf ~/.lmod.d/.cache
$ touch -d "24 hours ago" /home/user/moduleData/cacheDir/spiderT.luac_5.1 
$ module avail
### FAIL : SYSTEM SPIDER CACHE NOT USED : MODULES NOT YET IN CACHE ARE VISIBLE ! ###

$ rm -rf ~/.lmod.d/.cache
$ touch -d "23 hours ago" /home/user/moduleData/cacheDir/spiderT.luac_5.1 
$ module avail
### SUCCESS : SYSTEM SPIDER CACHE CACHE USED : ONLY MODULES IN CACHE ARE VISIBLE ! ###
```
I would like the cache never expires because I have a formal procedure for installing packages on my system that runs the `update_lmod_system_cache_files` script.

Since my cache file `/home/user/moduleData/cacheDir/spiderT.luac_5.1` is newer than the timestamp file `/home/user/moduleData/system.txt`, do you know why the cache expires after 24 hours ?

> If you have a formal procedure for installing packages on your system, then I recommend you to do the following. Have the install procedure run the update_lmod_system_cache_files script. This will create a file called “system.txt”, which marks the time that the system was last updated, so that Lmod knows that the cache is still good.

Reference : http://lmod.readthedocs.io/en/latest/130_spider_cache.html

> A cache file can have a system timestamp associated with it.  If it does then as long as the cache file is the same or newer then the timestamp then it is good forever.

Reference : https://github.com/TACC/Lmod/blob/master/src/Cache.lua

Thanks for you help !

**Comment by rtmclay** [comment]
Please re-run your experiment again with -D:
    module -D avail  2> avail.23.log
    module -D avail  2> avail.24.log

and see where the difference is between the two debug outputs.  If you want you can send me the two log files.

**Comment by rtmclay** [comment]
Thanks very much for the bug report!  The problem was that I didn't get the timestamp modify time correctly.  I have fixed the problem and uploaded the fixes as 7.7.15.  Please test it out to see if it works for you.

**Comment by guest1604** [comment]
Thank you a lot for this fix. Now, it works fine !
